### ISSUE TITLE
ArrayIndexOutOfBounds

### ISSUE THREAD

**Issue by sogriffin98** [s-andrews/FastQC#113]
Hello,

I am having an issue with fastqc throwing an out of bounds exception issue.

I ran the following line:
'fastqc -t 2 TET_P6_1_trated.fastq' 

It comes up with the following:
Started analysis of TET_P6_1_trated.fastq
Exception in thread "Thread-1" java.lang.ArrayIndexOutOfBoundsException: Index 65533 out of bounds for length 150
        at uk.ac.babraham.FastQC.Utilities.QualityCount.addValue(QualityCount.java:39)
        at uk.ac.babraham.FastQC.Modules.PerBaseQualityScores.processSequence(PerBaseQualityScores.java:148)
        at uk.ac.babraham.FastQC.Analysis.AnalysisRunner.run(AnalysisRunner.java:89)
        at java.base/java.lang.Thread.run(Thread.java:829)

I wonder if you can help?

Thank you

**Comment by s-andrews** [comment]
I had a dig into this.  The error you're getting comes from taking a character from the quality string and looking up its ASCII value.  This should always be a value <150 or so since it represents a phred score.

In your case you're getting 65533 which is the unicode fallback character (https://www.fileformat.info/info/unicode/char/fffd/index.htm) which is the value used when the character supplied doesn't conform to any known unicode character.

The bottom line for this is that your file is corrupt, and there's a really weird piece of data (not a normal letter) sitting in one of your quality strings.  The fix for this would be to recopy your file from where it was originally generated, or rerun the base calling if this is the primary copy.

Sorry.

**Comment by sogriffin98** [comment]
No worries! Thank you
